include/sea_dsa/ShadowMem.hh

```diff
@@ -30,7 +49,24 @@ public:
 	    llvm::Pass &pass /* for dominatorTree and assumptionCache*/,
 	    bool splitDsaNodes = false, bool computeReadMod = false,
 	    bool memOptimizer = true, bool useTBAA = true);
+  
   bool runOnModule(llvm::Module &M);
+
+  // Return the offset associated to cell c. The offset can be
+  // adjusted to zero if horn-sea-dsa-split=false.
+  unsigned getOffset(const Cell &c) const;
+
+  ShadowMemInstOp getShadowMemOp(const llvm::CallInst &ci) const; 
+  
+  // Return cell associated to the shadow mem call instruction.  It
+  // can be null if ci is not a shadow mem call instruction.
+  const Cell* getShadowMemCell(const llvm::CallInst &ci) const;
+
+  // Return a pair <def,use> with the variables which are defined
+  // and/or used by the shadow mem instruction. If the shadow
+  // instruction does not use or define a shadow variable then the
+  // corresponding pair element is null.
+  std::pair<llvm::Value*, llvm::Value*> getShadowMemVars(llvm::CallInst &ci) const;
```

Comment:
I believe that there are up 3 variables. One def and 2 uses. This is in case of `mem.transfer`.

Reply:
Why two uses? I was considering the source pointer as an use and the destination as a def

Reply:
`memcpy(dst, src)` defines `dst` and uses both `dst` and `src`. `dst` is used to connect the previous def of `dst` to this one.

Reply:
`memcpy(dst, src)` is translated into two memory shadow instructions:
- `mem.shadow.trfr.load(src)`
- `sh = mem.shadow.store(dst)`
so in the end all memory shadow instruction have at most one definition and one use.
We can still change the API to allow more than one use per shadow mem instruction in the future, what do you think?